### PR TITLE
APPS-1329 Fix marker error for listing backups

### PR DESCRIPTION
**Pull request by filkeith**
- fixed marker parameter, renamed to startAfter
- added flags docs

### CODE REVIEW

**Comment by filkeith** [comment]
After this merge, I'll update backup service

**Review by korotkov-aerospike — APPROVED** [review]

File: io/azure/blob/options.go [hunk 1]
```diff
@@ -30,14 +30,10 @@ type options struct {
...
-	// marker is a string value that identifies the portion of the list of containers to be
-	// returned with the next listing operation.
-	// The operation returns the NextMarker value within the response body if the listing
-	// operation did not return all containers remaining to be listed with the current page.
-	// The NextMarker value can be used
-	// as the value for the marker parameter in a subsequent call to request the next
-	// page of list items. The marker value is opaque to the client.
-	marker string
+	// startAfter is an artificial parameter. Used to skip objects in the storage.
+	// If it is set, then we compare the names received from the storage and if the name is less
```
Comment (reugn) [inline io/azure/blob/options.go:34]: lexicographically

File: io/azure/blob/options.go [hunk 2]
```diff
@@ -30,14 +30,10 @@ type options struct {
...
+	// than the specified parameter, we skip this object.
```
Comment (reugn) [inline io/azure/blob/options.go:35]: Mention that the boundary is exclusive.

File: io/azure/blob/reader.go [hunk 3]
```diff
@@ -121,7 +120,8 @@ func (r *Reader) streamDirectory(
 
 		for _, blob := range page.Segment.BlobItems {
 			// Skip files in folders.
-			if isDirectory(r.prefix, *blob.Name) && !r.withNestedDir {
+			if (isDirectory(r.prefix, *blob.Name) && !r.withNestedDir) ||
+				isSkippedByStartAfter(r.startAfter, *blob.Name) {
```
Comment (reugn) [inline io/azure/blob/reader.go:124]: It is better to extract the condition to a method and reuse.

File: io/azure/blob/options.go (hunk 1 above)
Comment (filkeith) [reply io/azure/blob/options.go]: added

File: io/azure/blob/options.go (hunk 2 above)
Comment (filkeith) [reply io/azure/blob/options.go]: added

File: io/azure/blob/reader.go (hunk 3 above)
Comment (filkeith) [reply io/azure/blob/reader.go]: fixed

**Review by reugn — APPROVED** [review]

**Pull request merged**